### DISCUSSION TITLE
How to call properly a function within a policy?

### DISCUSSION THREAD

**Discussion by javierguzman** [supabase/supabase#11628] [Questions] [answered]
Hello all,

I have a function called "is_user_participant_in_conversation" which receives a conversation ID as a parameter and returns a boolean.

I want to create a policy for messages, so users can only insert messages if it is a participant in the conversation.

In the CHECK field of the policy I have the following:

`((creator_id = uid()) AND is_user_participant_in_conversation(conversation_id))
`
However, when trying to insert a message it says that column conversation_id does not exist.

When I insert a message I use this:

```
const { data, error } = await supabase.from('message').insert({
    creator_id: creatorID,
    conversation_id: conversationID,
blabla
  });  

```
So I thought I could call my function is_user_participant_in_conversation with the conversation_id value that the new message has. However, it seems this is wrong.

So how can I achieve what I want? How can I call my function with the value of the message that is trying to be created?

Thank you in advance and regards

**Comment by GaryAustin1** [comment]
Is conversation_id a column in the table the policy is on?   If so you should have access to it, if not then no you can’t pass in extra data.

**Reply by javierguzman** [reply]
I found it at the end even though it was in front of my face all the time. It was a typo inside is_user_participant_in_conversation function, specifically, "WITH found_participants AS (select * from conversation where conversation_id=p_conversation_id" should have been "from participant"

**Comment by javierguzman** [comment, marked as answer]
I found it at the end even though it was in front of my face all the time. It was a typo inside is_user_participant_in_conversation function, specifically, "WITH found_participants AS (select * from conversation where conversation_id=p_conversation_id" should have been "from participant"
